forms/StyledSection/StyledSection.js

```diff
@@ -1,32 +1,92 @@
 import React from "react";
 import PropTypes from "prop-types";
+import styled from "styled-components";
 
 import Section from "../Section";
+import Heading from "../../composites/basic/Heading";
+import colors from "../../style-guide/colors.json";
+import { rgba } from "../../style-guide/helpers";
+import { Icon } from "../../composites/Plugin/Shared/components/Icon";
+import * as icons from "../../style-guide/svg";
+
+const StyledHeading = styled( Heading )``;
+
+const StyledIcon = styled( Icon )``;
+
+const StyledSectionBase = styled( Section )`
+	box-shadow: 0 1px 2px ${ rgba( colors.$color_black, 0.2 ) };
+	background-color: ${ colors.$color_white };
+	padding: 0 20px 16px;
+
+	*, & {
+		box-sizing: border-box;
+
+		&:before, &:after {
+			box-sizing: border-box;
+		}
+	}
+
+	& ${ StyledHeading } {
+		display: flex;
+		align-items: center;
+		padding: 8px 0 0;
+		font-size: 1rem;
+		line-height: 1.5;
+		margin: 0 0 16px;
+		font-family: "Open Sans", sans-serif;
+		font-weight: 300;
+		color: ${ colors.$color_grey_dark };
+	}
+
+	& ${ StyledIcon } {
+		margin-right: 8px;
+	}
+`;
 
 /**
- * Represents a visual section within the page.
+ * Creates a styled section within the page.
  *
- * @param {Object} props The props for this component.
- * @param {string} props.title The title to put inside the heading.
- * @param {string} props.icon The classname to give to the section element.
- * @param {ReactElement} props.sectionContent The content to put inside the section.
+ * @param {Object} props The props to use.
  * @returns {ReactElement} The rendered component.
- * @constructor
  */
-const StyledSection = ( { title, icon, sectionContent } ) => {
-	let className = "yoast-section";
-	let headingClassName = "yoast-section__heading yoast-section__heading-icon yoast-section__heading-icon-" + icon;
-
+const StyledSection = ( props ) => {
 	return (
-		<Section level={3} headingText={title} className={className} headingClassName={headingClassName} children={sectionContent} />
+		<StyledSectionBase
+			className={ props.className }
+		>
+			<StyledHeading
+				level={ props.headingLevel }
+				className={ props.headingClassName }
+			>
+				{ props.headingIcon &&
+					<StyledIcon
+						icon={ icons[ `${ props.headingIcon }` ] }
+						color={ props.headingIconColor }
+						size={ props.headingIconSize }
+					/>
+				}
+				{ props.headingText }
+			</StyledHeading>
+			{ props.sectionContent }
```

Comment:
This prop is unnecessary. I think all "regular" content should be passed through `props.children`. Allowing developers to achieve the same result with different approaches will cause inconsistent code.

Reply:
I'd agree, but this is the way content is currently passed on Premium in `premium/assets/js/src/keywordSuggestions/KeywordSuggestions.js` so we should make changes there too.